src/lsp-server.spec.ts

```diff
@@ -934,3 +934,41 @@ describe('diagnostics (no client support)', () => {
         assert.isUndefined(resultsForFile, 'Unexpected diagnostics received');
     }).timeout(10000);
 });
+
+describe('inlayHints', () => {
+    it('inlayHints', async () => {
+        const doc = {
+            uri: uri('module.ts'),
+            languageId: 'typescript',
+            version: 1,
+            text: `
+        export function foo() {
+          return 3
+        }
+      `
+        };
+        server.initialize({
+            initializationOptions: {
+                preferences: {
+                    includeInlayFunctionLikeReturnTypeHints: true
```

Comment:
This seems a bit sketchy  because that option doesn't seem to be part of the Typescript `UserPreferences` (https://github.com/microsoft/TypeScript/blob/59fb3731e3858743bd0aae82e35097ab4f4cc7df/src/server/protocol.ts#L3356-L3398)

Also the fact that you have to ignore the type issue in https://github.com/typescript-language-server/typescript-language-server/blob/ca191d2c4091a662dfffbf120e380eec61badcc7/src/tsp-client.spec.ts#L76-L76

Is that because Typescript exposed it in a sloppy way or there is a different way of setting that?

Reply:
The latter is true, the test fails without that line but you cannot set it because it's in something else, not fully UserPreferences. I'm not sure why typescript made it so ugly